### DISCUSSION TITLE
Authentication System

### DISCUSSION THREAD

**Discussion by Segelzwerg** [Whist-Team/Whist-Core#7] [Ideas]
Following #4

**Comment by Segelzwerg** [comment]
Shouldn't the responsibility of authenticating users be part of Whist-Server?

**Reply by iTitus** [reply]
The specifics (OAuth2 etc.) should be over there yes.
But I think the decision may have influence on the datastructures used to represent users and also the type of reference used in code/storage

**Reply by iTitus** [reply]
Ideally we can have a `Player` on the core side and a `User` on the server side.

**Reply by iTitus** [reply]
You might be thinking "why do we need to classes?".
This would of course be even better. But I think in that case we might have to include pydantic as dep in the core project.

-> https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/

**Reply by Segelzwerg** [reply]
ok, that sounds reasonable. I think token-based authentication is a good approach and easy to implement.
